tests/test/system/test_exec_graph.cpp

```diff
@@ -0,0 +1,90 @@
+#include <catch2/catch.hpp>
+
+#include "faasm_fixtures.h"
+
+#include <faabric/runner/FaabricMain.h>
+#include <faabric/scheduler/ExecGraph.h>
+#include <faabric/scheduler/Scheduler.h>
+// DELETE MEEEE
+#include <faabric/util/logging.h>
+#include <faaslet/Faaslet.h>
+
+namespace tests {
+class ExecGraphTestFixture : public FunctionExecTestFixture
```

Comment:
We already have tests for the execution graph in `tests/test/scheduler/test_exec_graph.cpp`, I think these can just be put in there can't they? The exec graph doesn't really have any thing to do with the `system` module.

Reply:
Those tests live in `faabric`, and in `faabric` we [already check](https://github.com/faasm/faabric/blob/f7a910bcb262777ca22b47646b7c05497bc63f44/tests/test/scheduler/test_mpi_exec_graph.cpp#L46) that if the flag is not set, the execution graph is not generated.

I thought your comment in the previous review meant that you wanted to have an e2e check in faasm.

Reply:
Gah sorry I'm getting confused now 😵, you're right, difficult to merge these tests with those in a different repo 😂 .

However, I still think we can find a better home for them. I think the first generic one can go into `tests/test/faaslet/test_chaining.cpp`, then the MPI one into `tests/test/faaslet/test_mpi.cpp`.